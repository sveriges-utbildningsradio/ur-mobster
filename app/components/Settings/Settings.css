.buttonWrap {
  position: absolute;
  right: 40px;
  top: 40px;
  z-index: 2;
}

.settingsButtonWrapper {
  display: flex;
}

.settingsButton {
  height: 24px;
  margin-left: 12px;
  width: 24px;
}

.pageWrap {
  align-items: center;
  background-color: #110737;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  left: 0;
  position: absolute;
  top: 0;
  width: 100vw;
  z-index: 1;
}

.innerWrap {
  width: 950px;
}

.header {
  color: #fff;
  font-family: 'Exo 2', sans-serif;
  font-size: 55px;
  font-weight: 800;
  margin-bottom: 70px;
  letter-spacing: -0.82px;
  line-height: 0.58;
}

.settingsWrap {
  border-bottom: 2px solid #caf5ff;
  margin-bottom: 80px;
}

.settingWrap {
  align-items: center;
  border-top: 2px solid #caf5ff;
  display: grid;
  grid-template-columns: 80px auto 210px 190px;
  grid-column-gap: 60px;
  max-width: 927px;
  padding: 30px 0 30px 100px;
}

.icon {
  width: 62px;
}

.settingLabel {
  border-right: 2px solid #caf5ff;
  color: #fff;
  font-family: 'Roboto', sans-serif;
  font-size: 36px;
  text-align: left;
}
