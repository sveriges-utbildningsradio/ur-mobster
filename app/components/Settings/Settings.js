// @flow
import React, { useContext, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import styles from './Settings.css'
import Cog from '../../assets/cog.png'
import Close from '../../assets/close.png'
import Time from '../../assets/time.png'
import Break from '../../assets/break.png'
import Language from '../../assets/language.png'
import DefaultButton from '../Button/DefaultButton'
import SettingsToggle from './components/SettingsToggle'
import SettingsDuration from './components/SettingsDuration/SettingsDuration'
import { SettingsStoreContext } from '../../store/store'

export const LANGUAGE_OPTIONS = [
  {
    label: 'Svenska',
    value: 'sv'
  },
  {
    label: 'English',
    value: 'en'
  }
]

export const Settings = () => {
  const {
    duration,
    breakDuration,
    breakFrequency,
    language,
    updateDuration,
    updateBreakDuration,
    updateBreakFrequency
  } = useContext(SettingsStoreContext)
  const [showingSettings, toggleShowingSettings] = useState(false)

  return (
    <>
      <button
        className={styles.buttonWrap}
        onClick={() => toggleShowingSettings(!showingSettings)}
        type="button"
      >
        {showingSettings ? (
          <span className={styles.settingsButtonWrapper}>
            <FormattedMessage id="close" />
            <img
              alt="Close settings"
              className={styles.settingsButton}
              src={Close}
            />
          </span>
        ) : (
          <img
            alt="Settings page"
            className={styles.settingsButton}
            src={Cog}
          />
        )}
      </button>
      {!!showingSettings && (
        <div className={styles.pageWrap}>
          <div className={styles.innerWrap}>
            <h2 className={styles.header}>
              <FormattedMessage id="settingsHeader" />
            </h2>
            <ul className={styles.settingsWrap}>
              <li className={styles.settingWrap}>
                <img
                  alt="Duration settings"
                  className={styles.icon}
                  src={Time}
                />
                <span className={styles.settingLabel}>
                  <FormattedMessage id="time" />
                </span>
                <SettingsDuration time={duration} updaterFn={updateDuration} />
              </li>
              <li className={styles.settingWrap}>
                <img alt="Pause settings" className={styles.icon} src={Break} />
                <span className={styles.settingLabel}>
                  <FormattedMessage id="break" />
                </span>
                <FormattedMessage id="length">
                  {msg => (
                    <SettingsDuration
                      prefixLabel={`${msg} `}
                      time={breakDuration}
                      updaterFn={updateBreakDuration}
                    />
                  )}
                </FormattedMessage>

                <FormattedMessage id="interval">
                  {msg => (
                    <SettingsDuration
                      prefixLabel={`${msg} `}
                      time={breakFrequency}
                      updaterFn={updateBreakFrequency}
                    />
                  )}
                </FormattedMessage>
              </li>
              <li className={styles.settingWrap}>
                <img
                  alt="Language settings"
                  className={styles.icon}
                  src={Language}
                />
                <span className={styles.settingLabel}>
                  <FormattedMessage id="language" />
                </span>
                <SettingsToggle active={language} options={LANGUAGE_OPTIONS} />
              </li>
            </ul>
            <DefaultButton
              handleClick={() => toggleShowingSettings(false)}
              textId="doneButton"
            />
          </div>
        </div>
      )}
    </>
  )
}

export default Settings
