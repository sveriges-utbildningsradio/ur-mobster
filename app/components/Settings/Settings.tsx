import React, { useContext, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import styles from './Settings.css'
import Cog from '../../assets/cog.png'
import Close from '../../assets/close.png'
import Time from '../../assets/time.png'
import Break from '../../assets/break.png'
import Language from '../../assets/language.png'
import DefaultButton from '../Button/DefaultButton'
import SettingsToggle from './components/SettingsToggle'
import SettingsDuration from './components/SettingsDuration/SettingsDuration'
import { SettingsStoreContext } from '../../store/store'

export const LANGUAGE_OPTIONS = [
  {
    label: 'Svenska',
    value: 'sv'
  },
  {
    label: 'English',
    value: 'en'
  }
]

const Settings = () => {
  const {
    duration,
    breakDuration,
    breakFrequency,
    language,
    updateDuration,
    updateBreakDuration,
    updateBreakFrequency
  } = useContext(SettingsStoreContext)
  const [showingSettings, toggleShowingSettings] = useState(false)

  return (
    <>
      <button
        className={styles.buttonWrap}
        onClick={() => toggleShowingSettings(!showingSettings)}
        data-e2e="settings-toggle-button"
        type="button"
      >
        {showingSettings ? (
          <span className={styles.settingsButtonWrapper}>
            <FormattedMessage id="close" />
            <img
              alt="Close settings"
              className={styles.settingsButton}
              src={Close}
            />
          </span>
        ) : (
          <img
            alt="Settings page"
            className={styles.settingsButton}
            src={Cog}
          />
        )}
      </button>
      {!!showingSettings && (
        <div className={styles.pageWrap}>
          <div>
            <div className={styles.innerWrap}>
              <h2 className={styles.header} data-e2e="settings-header">
                <FormattedMessage id="settingsHeader" />
              </h2>
              <ul className={styles.settingsWrap}>
                <li className={styles.settingWrap}>
                  <img
                    alt="Duration settings"
                    className={styles.icon}
                    src={Time}
                  />
                  <span className={styles.settingLabel}>
                    <FormattedMessage id="time" />
                  </span>
                  <SettingsDuration
                    dataE2EDecrease="decrease-duration"
                    dataE2EIncrease="increase-duration"
                    time={duration}
                    updaterFn={updateDuration}
                    updateByMinute
                  />
                </li>
                <li className={styles.settingWrap}>
                  <img
                    alt="Pause settings"
                    className={styles.icon}
                    src={Break}
                  />
                  <span className={styles.settingLabel}>
                    <FormattedMessage id="break" />
                  </span>
                  <FormattedMessage id="length">
                    {msg => (
                      <SettingsDuration
                        dataE2EDecrease="decrease-break-duration"
                        dataE2EIncrease="increase-break-duration"
                        prefixLabel={`${msg} `}
                        time={breakDuration}
                        updaterFn={updateBreakDuration}
                      />
                    )}
                  </FormattedMessage>

                  <FormattedMessage id="interval">
                    {msg => (
                      <SettingsDuration
                        dataE2EDecrease="decrease-break-frequency"
                        dataE2EIncrease="increase-break-frequency"
                        prefixLabel={`${msg} `}
                        time={breakFrequency}
                        updaterFn={updateBreakFrequency}
                      />
                    )}
                  </FormattedMessage>
                </li>
                <li className={styles.settingWrap}>
                  <img
                    alt="Language settings"
                    className={styles.icon}
                    src={Language}
                  />
                  <span className={styles.settingLabel}>
                    <FormattedMessage id="language" />
                  </span>
                  <SettingsToggle
                    active={language}
                    options={LANGUAGE_OPTIONS}
                  />
                </li>
              </ul>
              <DefaultButton
                handleClick={() => toggleShowingSettings(false)}
                textId="doneButton"
              />
            </div>
          </div>
          <FormattedMessage
            id="createdBy"
            values={{
              andre: <a href="https://github.com/Aelmoznino">@Aelmoznino</a>,
              agnes: <a href="https://github.com/agnesforsell">@AgnesForsell</a>
            }}
          />
        </div>
      )}
    </>
  )
}

export default Settings
