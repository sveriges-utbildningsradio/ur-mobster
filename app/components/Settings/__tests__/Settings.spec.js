import React from 'react'
import { fireEvent, cleanup } from '@testing-library/react'
import 'jest-dom/extend-expect'
import renderWithReactIntl from '../../../utils/testHelpers'
import Settings from '../Settings'

describe('components/Settings', () => {
  afterEach(cleanup)

  it('renders just the button when closed', () => {
    const { container } = renderWithReactIntl(<Settings />)

    expect(container).toMatchSnapshot()
  })

  it('renders when open', () => {
    const { container, getByAltText } = renderWithReactIntl(<Settings />)

    fireEvent.click(getByAltText('Settings page'))
    expect(container).toMatchSnapshot()
  })

  it('closes when clicking the Done button', () => {
    const { container, getByAltText, getByText } = renderWithReactIntl(
      <Settings />
    )

    fireEvent.click(getByAltText('Settings page'))

    fireEvent.click(getByText('Klar!'))

    expect(container).toMatchSnapshot()
  })
})
