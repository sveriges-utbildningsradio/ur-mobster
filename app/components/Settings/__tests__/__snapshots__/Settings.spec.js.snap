// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Settings closes when clicking the Done button 1`] = `
<div>
  <button
    class="buttonWrap"
    data-e2e="settings-toggle-button"
    type="button"
  >
    <img
      alt="Settings page"
      class="settingsButton"
      src="test-file-stub"
    />
  </button>
</div>
`;

exports[`components/Settings renders just the button when closed 1`] = `
<div>
  <button
    class="buttonWrap"
    data-e2e="settings-toggle-button"
    type="button"
  >
    <img
      alt="Settings page"
      class="settingsButton"
      src="test-file-stub"
    />
  </button>
</div>
`;

exports[`components/Settings renders when open 1`] = `
<div>
  <button
    class="buttonWrap"
    data-e2e="settings-toggle-button"
    type="button"
  >
    <span
      class="settingsButtonWrapper"
    >
      <span>
        Stäng
      </span>
      <img
        alt="Close settings"
        class="settingsButton"
        src="test-file-stub"
      />
    </span>
  </button>
  <div
    class="pageWrap"
  >
    <div>
      <div
        class="innerWrap"
      >
        <h2
          class="header"
          data-e2e="settings-header"
        >
          <span>
            Mobster-inställningar
          </span>
        </h2>
        <ul
          class="settingsWrap"
        >
          <li
            class="settingWrap"
          >
            <img
              alt="Duration settings"
              class="icon"
              src="test-file-stub"
            />
            <span
              class="settingLabel"
            >
              <span>
                Tid
              </span>
            </span>
            <div
              class="wrap"
            >
              <p
                class="time"
              >
                10
                 min
              </p>
              <div
                class="buttonsWrap"
              >
                <button
                  class="button"
                  data-e2e="increase-duration"
                  type="button"
                >
                  <img
                    alt="Increase time 5 minutes"
                    class="icon"
                    src="test-file-stub"
                  />
                </button>
                <button
                  class="button"
                  type="button"
                >
                  <img
                    alt="Decrease time 5 minutes"
                    class="icon"
                    data-e2e="decrease-duration"
                    src="test-file-stub"
                  />
                </button>
              </div>
            </div>
          </li>
          <li
            class="settingWrap"
          >
            <img
              alt="Pause settings"
              class="icon"
              src="test-file-stub"
            />
            <span
              class="settingLabel"
            >
              <span>
                Rast
              </span>
            </span>
            <div
              class="wrap"
            >
              <p
                class="time"
              >
                Längd 
                5
                 min
              </p>
              <div
                class="buttonsWrap"
              >
                <button
                  class="button"
                  data-e2e="increase-break-duration"
                  type="button"
                >
                  <img
                    alt="Increase time 5 minutes"
                    class="icon"
                    src="test-file-stub"
                  />
                </button>
                <button
                  class="button"
                  type="button"
                >
                  <img
                    alt="Decrease time 5 minutes"
                    class="icon"
                    data-e2e="decrease-break-duration"
                    src="test-file-stub"
                  />
                </button>
              </div>
            </div>
            <div
              class="wrap"
            >
              <p
                class="time"
              >
                Intervall 
                50
                 min
              </p>
              <div
                class="buttonsWrap"
              >
                <button
                  class="button"
                  data-e2e="increase-break-frequency"
                  type="button"
                >
                  <img
                    alt="Increase time 5 minutes"
                    class="icon"
                    src="test-file-stub"
                  />
                </button>
                <button
                  class="button"
                  type="button"
                >
                  <img
                    alt="Decrease time 5 minutes"
                    class="icon"
                    data-e2e="decrease-break-frequency"
                    src="test-file-stub"
                  />
                </button>
              </div>
            </div>
          </li>
          <li
            class="settingWrap"
          >
            <img
              alt="Language settings"
              class="icon"
              src="test-file-stub"
            />
            <span
              class="settingLabel"
            >
              <span>
                Språk
              </span>
            </span>
            <ul
              class="sc-bdVaJa imDAUP"
            >
              <li
                class="sc-bwzfXH lkHdtP"
                data-e2e="settings-toggle-sv"
              >
                Svenska
                <img
                  class="sc-htpNat cfdZzv"
                  src="test-file-stub"
                />
              </li>
              <li
                class="sc-bwzfXH akcDY"
                data-e2e="settings-toggle-en"
              >
                English
              </li>
            </ul>
          </li>
        </ul>
        <button
          class="defaultButton"
          data-e2e="doneButton"
          type="button"
        >
          <span>
            Klar!
          </span>
        </button>
      </div>
    </div>
    <span>
      Skapat av 
      <a
        href="https://github.com/Aelmoznino"
      >
        @Aelmoznino
      </a>
       och 
      <a
        href="https://github.com/agnesforsell"
      >
        @AgnesForsell
      </a>
       för Sveriges Utbildningsradio AB
    </span>
  </div>
</div>
`;
